pkg/workspace/controllers/workspace_controller.go

```diff
@@ -658,68 +658,43 @@ func (c *WorkspaceReconciler) applyInference(ctx context.Context, wObj *kaitov1b
 		} else if wObj.Inference != nil && wObj.Inference.Preset != nil {
 			presetName := string(wObj.Inference.Preset.Name)
 			model := plugin.KaitoModelRegister.MustGet(presetName)
-
 			inferenceParam := model.GetInferenceParameters()
+			revisionStr := wObj.Annotations[kaitov1beta1.WorkspaceRevisionAnnotation]
+
+			var workloadObj client.Object
+			workloadObj, err = inference.GeneratePresetInference(ctx, wObj, revisionStr, model, c.Client)
```

Comment:
There is a behavior change here. With line 665, any changes that user applied to the deployment podspec directly will be overwritten. For now, we still allow users to change the deployment spec (such as command line). We will only update the fields that user can change via the workspace cr.

Reply:
Addressed this in the latest commit